### PR TITLE
Happy Thoughts API

### PR DESCRIPTION
**Pull request by eilsson**
Heroku: https://happy-thoughts-emmie.herokuapp.com/
Netlify: https://happy-thoughts-emmie.netlify.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
 import mongoose from 'mongoose'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false })
```
Comment (JennieDalgren) [inline server.js]: 😎

File: server.js [hunk 2]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
...
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
```
Comment (JennieDalgren) [inline server.js]: ⭐️

File: server.js [hunk 3]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
...
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', async (req, res) => {
+  const sortQuery = req.query.sort
+
+  const sortByQuery = () => {
```
Comment (JennieDalgren) [inline server.js]: nice handling

File: server.js [hunk 4]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
...
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
```
Comment (JennieDalgren) [inline server.js]: Also add the error handling for the database connection. Check Damiens video about setting that middleware up here:
https://technigo.wistia.com/medias/l5wqrcu4to?wtime=9m22s

File: server.js [hunk 5]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
...
+  try {
+    await newThought.save()
+    res.status(201).json(newThought)
+  } catch (err) {
+    res.status(400).json({ message: 'Could not save thought', error: err.errors })
+  }
+})
+
+app.post('/:thoughtId/like', async (req, res) => {
+  const thoughtId = req.params.thoughtId
```
Comment (JennieDalgren) [inline server.js]: you can destructure here as well, to be consistent.
`const { thoughtId } = req.params`

File: server.js [hunk 6]
```diff
@@ -4,26 +4,85 @@ import cors from 'cors'
...
+
+  try {
+    const likedThought = await Thought.findByIdAndUpdate(thoughtId, { $inc: { hearts: 1 } }, { new: true })
+    res.status(201).json(likedThought)
+  } catch (err) {
+    if (err.kind === "ObjectId") {
```
Comment (JennieDalgren) [inline server.js]: ⭐️

File: server.js (hunk 4 above)
Comment (eilsson) [reply server.js]: Thank you, Jennier. I have added the middleware.

File: server.js (hunk 5 above)
Comment (eilsson) [reply server.js]: Thanks, I've changed that now.

**Pull request closed without merging**